sdk/signalr/Microsoft.Azure.WebJobs.Extensions.SignalRService/src/SignalROptions.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Core.Serialization;
+using Microsoft.Azure.SignalR;
+using Microsoft.Azure.SignalR.Management;
+using Microsoft.Azure.WebJobs.Hosting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    /// <summary>
+    /// Configuration options for SignalR extensions.
+    /// </summary>
+    /// <remarks>
+    /// Most properties here can be specified in configuration, except JSON serialization.
+    /// </remarks>
+    public class SignalROptions : IOptionsFormatter
+    {
+        public IList<ServiceEndpoint> ServiceEndpoints { get; } = new List<ServiceEndpoint>();
+
+        /// <summary>
+        /// Gets or sets the service transport type.
+        /// </summary>
+        public ServiceTransportType? ServiceTransportType { get; set; }
```

Comment:
Do we need to make this nullable?

Reply:
If it is nullable, users can distinguish whether they have set this value from the options printing result.